vault/resource_auth_backend.go

```diff
@@ -51,118 +51,136 @@ func AuthBackendResource() *schema.Resource {
+			config.IdentityTokenKey = v.(string)
+		}
+	}
+
 	options := &api.EnableAuthOptions{
 		Type:        mountType,
-		Description: d.Get("description").(string),
-		Local:       d.Get("local").(bool),
+		Description: d.Get(consts.FieldDescription).(string),
+		Local:       d.Get(consts.FieldLocal).(bool),
+		Config:      *config,
 	}
 
 	log.Printf("[DEBUG] Writing auth %q to Vault", path)
-	if err := client.Sys().EnableAuthWithOptions(path, options); err != nil {
-		return fmt.Errorf("error writing to Vault: %s", err)
+	if err := client.Sys().EnableAuthWithOptionsWithContext(ctx, path, options); err != nil {
+		return diag.Errorf("error writing to Vault: %s", err)
 	}
 
 	d.SetId(path)
 
-	return authBackendUpdate(d, meta)
+	return authBackendUpdate(ctx, d, meta)
 }
 
-func authBackendDelete(d *schema.ResourceData, meta interface{}) error {
+func authBackendDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
 	path := d.Id()
 
 	log.Printf("[DEBUG] Deleting auth %s from Vault", path)
 
-	if err := client.Sys().DisableAuth(path); err != nil {
-		return fmt.Errorf("error disabling auth from Vault: %s", err)
+	if err := client.Sys().DisableAuthWithContext(ctx, path); err != nil {
+		return diag.Errorf("error disabling auth from Vault: %s", err)
 	}
 
 	return nil
 }
 
-func authBackendRead(d *schema.ResourceData, meta interface{}) error {
+func authBackendRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
 	path := d.Id()
 
-	mount, err := mountutil.GetAuthMount(context.Background(), client, path)
+	mount, err := mountutil.GetAuthMount(ctx, client, path)
 	if errors.Is(err, mountutil.ErrMountNotFound) {
 		log.Printf("[WARN] Mount %q not found, removing from state.", path)
 		d.SetId("")
 		return nil
 	}
 
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	if err := d.Set("type", mount.Type); err != nil {
-		return err
+	if err := d.Set(consts.FieldType, mount.Type); err != nil {
+		return diag.FromErr(err)
 	}
 	if err := d.Set(consts.FieldPath, path); err != nil {
-		return err
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldDescription, mount.Description); err != nil {
+		return diag.FromErr(err)
 	}
-	if err := d.Set("description", mount.Description); err != nil {
-		return err
+	if err := d.Set(consts.FieldLocal, mount.Local); err != nil {
+		return diag.FromErr(err)
 	}
-	if err := d.Set("local", mount.Local); err != nil {
-		return err
+	if err := d.Set(consts.FieldAccessor, mount.Accessor); err != nil {
+		return diag.FromErr(err)
 	}
-	if err := d.Set("accessor", mount.Accessor); err != nil {
-		return err
+	if err := d.Set(consts.FieldIdentityTokenKey, mount.Config.IdentityTokenKey); err != nil {
```

Comment:
We can comment this out for now and add a TODO to add these back in. As of the current Vault API, the `identity_token_key` is not being returned back on a read from Vault on the read mount endpoint. This will be fixed in the next Vault API release, but until then we won't be able to set the Token Key from the Vault response. This may cause a drift